lte/gateway/c/core/oai/tasks/mme_app/mme_app_itti_messaging.c

```diff
@@ -143,8 +143,19 @@ status_code_e mme_app_send_s11_release_access_bearers_req(
   release_access_bearers_request_p =
       &message_p->ittiMsg.s11_release_access_bearers_request;
   release_access_bearers_request_p->local_teid = ue_mm_context->mme_teid_s11;
-  pdn_context_t* pdn_connection = ue_mm_context->pdn_contexts[pdn_index];
-  release_access_bearers_request_p->teid = pdn_connection->s_gw_teid_s11_s4;
+  /* Fetch the 1st PDN context to check if the message should be sent
+   * to spgw/sgw_s8 task
+   */
+  pdn_context_t* pdn_connection = ue_mm_context->pdn_contexts[0];
+  if (!pdn_connection) {
+    OAILOG_ERROR_UE(
+        LOG_MME_APP, ue_mm_context->emm_context._imsi64,
+        "PDN context is NULL for "
+        "ue id " MME_UE_S1AP_ID_FMT "\n",
+        ue_mm_context->mme_ue_s1ap_id);
+    OAILOG_FUNC_RETURN(LOG_MME_APP, RETURNerror);
+  }
+
```

Comment:
These changes would create issues in the use cases where different pdn sessions are targeted for different PGWs (e.g., IMS session is treated as inbound vs Internet session is treated as home). @uri200, @rsarwad  currently we do not have this session splitting feature for inbound roaming, but this would be one place where this accounting needs to be done if we want to support this feature. @pruthvihebbani it looks like a low hanging fruit to simply iterate over PDNs and use flags to send the request to each destination only once?

Reply:
when the response from a task is received, since we know the sending task, we can match against the routing info to delete the bearers at mme_app side.

Reply:
@ulaskozat After discussing with @ssanadhya and @rsarwad offline came up with the below logic to send RAB:
Traverse through all the PDN contexts to check if route_s11_messages_to_s8_task is set for at least 1 PDN. If set, send Release access bearer to sgw_s8 task. If route_s11_messages_to_s8_task is not set for at least 1 PDN, send 
Release access bearer to sgw task. Maintain a counter in ue_mm_context to track the Release access bearer requests sent/ Release access bearer rsps received. After receiving the all the Release access bearer responses, send UE context release cmd to s1ap task.

Reply:
sounds good.